src/libs/actions/IOU.js

```diff
@@ -1547,12 +1557,29 @@ function navigateToNextPage(iou, iouType, reportID, report) {
                       .filter((accountID) => currentUserAccountID !== accountID)
                       .map((accountID) => ({accountID, selected: true}))
                       .value();
-            setMoneyRequestParticipants(participants);
+            promises.push(setMoneyRequestParticipants(participants));
         }
-        Navigation.navigate(ROUTES.getMoneyRequestConfirmationRoute(iouType, reportID));
+        Promise.all(promises).then(() => {
```

Comment:
Similar comment here. Pushing all these merge promises into the array has made this code harder to reason about I think. We should document every reason why we will need to await an `Onyx.merge()` call.